src/args.rs

```diff
@@ -149,41 +149,40 @@ impl TemplatePath {
     /// # Panics
     /// Will panic if no path to a template has been set at all,
     /// which is never if Clap is initialized properly.
-    pub fn any_path(&self) -> String {
+    pub fn any_path(&self) -> &String {
         self.git
             .as_ref()
             .or(self.path.as_ref())
             .or(self.favorite.as_ref())
             .or(self.auto_path.as_ref())
-            .cloned()
             .unwrap()
     }
 
-    pub fn git(&self) -> Option<String> {
-        self.git.clone()
+    pub const fn git(&self) -> Option<&String> {
```

Comment:
nit: I would prefer to go with `&str` like `self.git.as_str()`

Reply:
nopez.... to do that requires use of `.map(|s| s.as_ref())`, and then there's issues on the calling side because it actually call other fn's that expect `Option<&String>`

Could then make it into `Cow<>` but is it worth it?

Reply:
it's possible to annotate the lifetime, you can find an [example here](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=e1de171af6a240a85f5a26c88ff7d414)

Reply:
let me put a shameless plug here, my latest blog post on `AsRef<>`:
https://d34dl0ck.me/rust-bites-as-ref/index.html

Reply:
we're actually in [this situation](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=b599397c6bf258f3c0a50a44ddc6bb43).

As I said, I don't know whether it's worth it - its way harder to read, and all the callsites actually expect `Option<String>`, so to really gain anything, we should also change that code... now it starts to grow into its own PR.

Reply:
It's ok, we don't need to go down this rabbit hole

Reply:
@sassman, as stated, don't know whether it was worth it...